server/server_loadbalancer.go

```diff
@@ -410,11 +410,29 @@ func buildHealthCheckOptions(lb healthcheck.BalancerHandler, backend string, hc
 		}
 	}
 
+	timeout := time.Duration(hcConfig.Timeout)
+	if hc.Timeout != "" {
+		timeoutOverride, err := time.ParseDuration(hc.Timeout)
+		if err != nil {
+			log.Errorf("Illegal health check timeout for backend '%s': %s", backend, err)
+		} else if timeoutOverride <= 0 {
+			log.Errorf("Health check timeout smaller than zero for backend '%s', backend", backend)
+		} else {
+			timeout = timeoutOverride
+		}
+	}
+
+	if timeout >= interval {
+		interval = timeout + time.Duration(1*time.Second)
```

Comment:
WDYT about just having a warn log and not changing the interval value. I think the +1s is a little bit arbitrary.

Reply:
We could do this but I'm unsure of the consequences of potentially having multiple health checks running at once. For example:

* Interval is set to 5s
* Timeout is set to 20s
* Backend is taking 15s to respond

In the above case we'll have up to 3 health checks running against a backend at the same time. Is this acceptable?

Reply:
In fact, we can't launch new healthcheck if we are still waiting for the previous healthcheck response because the healthcheck is in the same goroutine as the interval.
https://github.com/containous/traefik/blob/fdf14cd101821b5ab0d5fb62b5ffaa489ad5eaab/healthcheck/healthcheck.go#L125

Reply:
OK - I've removed the code that changes the interval and changed the log to warn level.